src/subscription/mapAsyncIterator.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2017, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+import { $$asyncIterator, getAsyncIterator } from 'iterall';
+
+/**
+ * Given an AsyncIterable and a callback function, return an AsyncIterator
+ * which produces values mapped via calling the callback function.
+ */
+export default function mapAsyncIterator<T, U>(
+  iterable: AsyncIterable<T>,
+  callback: (value: T) => U
+): AsyncIterator<U> {
+  // Fixes a temporary issue with Regenerator/Babel
+  // https://github.com/facebook/regenerator/pull/290
+  const iterator = iterable.next ? (iterable: any) : getAsyncIterator(iterable);
```

Comment:
I believe `regenerator-transform` and `babel-plugin-transform-regenerator` automatically include the latest point releases of `regenerator`, but explicitly bumping their minimum required version would be a good idea to help close the gap on this bug